periodic_job.go

```diff
@@ -0,0 +1,189 @@
+
+func (s *periodicIntervalSchedule) Next(t time.Time) time.Time {
+	return t.Add(s.interval)
+}
+
+// PeriodicJobBundle is a bundle of currently configured periodic jobs. It's
+// made accessible through Client, where new periodic jobs can be configured,
+// and only ones removed.
+type PeriodicJobBundle struct {
+	periodicJobEnqueuer *maintenance.PeriodicJobEnqueuer
+}
+
+func newPeriodicJobBundle(periodicJobEnqueuer *maintenance.PeriodicJobEnqueuer) *PeriodicJobBundle {
+	return &PeriodicJobBundle{
+		periodicJobEnqueuer: periodicJobEnqueuer,
+	}
+}
+
+// Adds a new periodic job to the client. The job is queued immediately if
+// RunOnStart is enabled, and then scheduled normally.
+//
+// Returns a periodic job handle which can be used to subsequently remove the
+// job if desired.
+//
+// Adding or removing periodic jobs has no effect unless this client is elected
+// leader because only the leader enqueues periodic jobs. To make sure that a
+// new periodic job is fully enabled or disabled, it should be added or removed
+// from _every_ active River client across all processes.
+func (b *PeriodicJobBundle) Add(periodicJob *PeriodicJob) rivertype.PeriodicJobHandle {
+	return b.periodicJobEnqueuer.Add(b.toInternal(periodicJob))
+}
+
+// AddMany adds many new periodic jobs to the client. The jobs are queued
+// immediately if their RunOnStart is enabled, and then scheduled normally.
+//
+// Returns a periodic job handle which can be used to subsequently remove the
+// job if desired.
+//
+// Adding or removing periodic jobs has no effect unless this client is elected
+// leader because only the leader enqueues periodic jobs. To make sure that a
+// new periodic job is fully enabled or disabled, it should be added or removed
+// from _every_ active River client across all processes.
+func (b *PeriodicJobBundle) AddMany(periodicJobs []*PeriodicJob) []rivertype.PeriodicJobHandle {
+	return b.periodicJobEnqueuer.AddMany(sliceutil.Map(periodicJobs, b.toInternal))
+}
+
+// Clear clears all periodic jobs, cancelling all scheduled runs.
+//
+// Adding or removing periodic jobs has no effect unless this client is elected
+// leader because only the leader enqueues periodic jobs. To make sure that a
+// new periodic job is fully enabled or disabled, it should be added or removed
+// from _every_ active River client across all processes.
+func (b *PeriodicJobBundle) Clear() {
+	b.periodicJobEnqueuer.Clear()
+}
+
+// Remove removes a periodic job, cancelling all scheduled runs.
+//
+// Requires the use of the periodic job handle that was returned when the job
+// was added.
+//
+// Adding or removing periodic jobs has no effect unless this client is elected
+// leader because only the leader enqueues periodic jobs. To make sure that a
+// new periodic job is fully enabled or disabled, it should be added or removed
+// from _every_ active River client across all processes.
+func (b *PeriodicJobBundle) Remove(periodicJobHandle rivertype.PeriodicJobHandle) {
```

Comment:
Debated on whether this should return (1) nothing, (2) an error, (3) an OK boolean, or even (4) panic when a job handle isn't found.

Decided to go with no return value because I figure an unknown job will be an edge most people don't have to handle, and it mirrors the Go API of `delete(map, key)`.